addons/web/static/src/core/checkbox/checkbox.js

```diff
@@ -48,13 +49,14 @@ export class CheckBox extends Component {
         if (!this.props.disabled) {
             ev.stopPropagation();
             input.checked = !input.checked;
-            this.props.onChange(input.checked);
+            this.props.onChange(input.checked, ev.shiftKey);
         }
     }
 
     onChange(ev) {
         if (!this.props.disabled) {
-            this.props.onChange(ev.target.checked);
+            this.props.onChange(ev.target.checked, this.shiftKey);
+            this.shiftKey = false;
```

Comment:
the whole diff in this file should be removed IMHO

Instead you should make the list_renderer aware of the shiftKey status.

For instance with something similar to:
```js
useExternalListener(window, "keydown", (ev)=> { this.shiftKeyMode = ev.shiftKey; });
useExternalListener(window, "keyup", ()=> { this.shiftKeyMode = false; });
```